openapi-core/src/main/java/io/ballerina/openapi/core/generators/type/generators/AllOfRecordTypeGenerator.java

```diff
@@ -106,6 +110,10 @@ public TypeDescriptorNode generateTypeDescriptorNode() throws OASTypeGenExceptio
         // filtering as input. Has to use this assertion statement instead of `if` condition, because to avoid
         // unreachable else statement.
         List<Schema<?>> allOfSchemas = schema.getAllOf();
+        List<String> requiredFields = schema.getRequired();
+        if (Objects.isNull(requiredFields)) {
+            requiredFields = new ArrayList<>();
```

Comment:
What if we do this ?
```suggestion
            requiredFields = Collections.emptyList();
```